src/v/redpanda/admin_server.cc

```diff
@@ -177,6 +174,115 @@ ss::future<> admin_server::configure_listeners() {
     }
 }
 
+void admin_server::rearm_log_level_timer() {
+    _log_level_timer.cancel();
+
+    auto next = std::min_element(
+      _log_level_resets.begin(),
+      _log_level_resets.end(),
+      [](const auto& a, const auto& b) {
+          return a.second.expires < b.second.expires;
+      });
+
+    if (next != _log_level_resets.end()) {
+        _log_level_timer.arm(next->second.expires);
+    }
+}
+
+void admin_server::log_level_timer_handler() {
+    for (auto it = _log_level_resets.begin(); it != _log_level_resets.end();) {
+        if (it->second.expires <= ss::timer<>::clock::now()) {
```

Comment:
I had a system like this in prod that eventually took down a deployment becuase i forgot to turn it off. 

can we have the expires be a default of say 10minutes or smht like that.

some default is better than no default. the user can explicitly set it to 10 days or smth insanely large like that :D but we should have sane defaults so we don't accidentally kill a node w/ trace logging.

Reply:
added 10 minute default timeout